### DISCUSSION TITLE
My native module doesn't work

### DISCUSSION THREAD

**Discussion by jihoobyeon** [microsoft/react-native-windows#10676] [Q&A] [answered]
Hello,\
I [made a native module](https://github.com/jihoobyeon/react-native-file-panel) for file picker, self-advised by those:\
https://microsoft.github.io/react-native-windows/docs/native-modules \
https://github.com/microsoft/react-native-windows-samples/tree/main/samples/NativeModuleSample/cppwinrt \
https://github.com/microsoft/Windows-universal-samples/tree/main/Samples/FilePicker

but when I ran my app, it says `Cannot read properties of undefined`\. no more error informations.
note that I'm **not** well experienced at C++, can you guys see my code and find whether what I did wrong or it is a bug?

**Comment by jihoobyeon** [comment]
My module haven't autolinked. so I had to manually link it.
however, this made another issue: https://github.com/microsoft/react-native-windows/issues/10684

**Comment by jbcullis** [comment]
Here are upload and export samples - https://github.com/jbcullis/react-native-windows-fs

Nb. These aren't generic so you'll need to make some adjustments based on your needs.

**Comment by jonthysell** [comment, marked as answer]
It looks like you've created the native project on your own, so you're not referencing any of the Microsoft.ReactNative dependencies you need to build a native module.

Please read https://microsoft.github.io/react-native-windows/docs/native-modules-setup#creating-a-new-native-module-library-project to see how you can use `react-native-windows-init` to set up your project with the correct dependencies.
